src/ic/libraries/src/indexer/mod.rs

```diff
@@ -0,0 +1,239 @@
+use crate::{
+    crypto::{
+        ethereum::recover_address_from_eth_signature,
+        streamr::{derive_event_model, types::JSONPayload},
+    },
+    remittance::types::{DataModel, Event, RemittanceSubscriber, Subscriber},
+};
+use candid::{CandidType, Principal};
+use ic_cdk::{api::call::RejectionCode, id};
+use serde_derive::Deserialize;
+use serde_json::Value;
+use std::{cell::RefCell, collections::BTreeMap};
+
+thread_local! {
+    pub static REMITTANCE_CANISTER: RefCell<Option<RemittanceSubscriber>> = RefCell::default();
+}
+
+pub mod permissions;
+pub type SubscriberStore = BTreeMap<Principal, Subscriber>;
+#[derive(Clone, Debug, CandidType, Deserialize)]
+pub struct Account {
+    pub balance: u64,
+}
+
+/// Set a value for the remittance canister registered
+pub fn set_remittance_canister(remittance_principal: Principal) {
```

Comment:
The `indexer` lib will likely not default behaviour to expect subscription for a remittance canister.
The `indexer` and `remittance` libs should useable inside of the same canister if it's suits the developer leveraging this tech.

It might even wiser for us to modularise the process of managing subscriptions from external cansiters.